### DISCUSSION TITLE
Any update coming for ergo soon

### DISCUSSION THREAD

**Discussion by jayintoledo** [coinfoundry/miningcore#895] [Ideas]
I would love to see a update for ergo... Any plans in the works?

**Comment by Konstantin35** [comment]
> 
> 
> I would love to see a update for ergo... Any plans in the works?

Look carefully if you are familiar with node ergo, you will understand that only the algorithm is added and that's it.

**Reply by jayintoledo** [reply]
i am familiar with ergo node... but not sure where you are referring the algo is added.. please clarify

**Comment by oliverw** [comment]
Ergo is fully implemented in a private build for a client that requested a period of semi-exclusive access (means CF can use it as well). The implementation will be merged into the main repository in a couple weeks.

**Reply by oliverw** [reply]
> lol, when it is private why you add it to coinfoundry.org ?

Read my previous post again.

**Reply by bdk773** [reply]
> Ergo is fully implemented in a private build for a client that requested a period of semi-exclusive access (means CF can use it as well). The implementation will be merged into the main repository in a couple weeks.

@oliverw was the implementation merged already with the main repo? Meaning are we able to add Ergo daemon or not yet? If not any eta that could be shared? Thank you.

**Reply by oliverw** [reply]
It is: https://github.com/coinfoundry/miningcore/releases/tag/v58

**Reply by bdk773** [reply]
> 
> 
> It is: https://github.com/coinfoundry/miningcore/releases/tag/v58

thank You @oliverw, any reference/example for the ergo pool.json configuration?

**Reply by oliverw** [reply]
> > It is: https://github.com/coinfoundry/miningcore/releases/tag/v58
> 
> thank You @oliverw, any reference/example for the ergo pool.json configuration?

There isn't really anything special about Ergo.

**Reply by bdk773** [reply]
> 
> 
> > > It is: https://github.com/coinfoundry/miningcore/releases/tag/v58
> > 
> > 
> > thank You @oliverw, any reference/example for the ergo pool.json configuration?
> 
> There isn't really anything special about Ergo.

Understood, in this case what (which coin) could I use as a reference then for that coin to build?

**Reply by oliverw** [reply]
> > > > It is: https://github.com/coinfoundry/miningcore/releases/tag/v58
> > > 
> > > 
> > > thank You @oliverw, any reference/example for the ergo pool.json configuration?
> > 
> > 
> > There isn't really anything special about Ergo.
> 
> Understood, in this case what (which coin) could I use as a reference then for that coin to build?

I'd say any of the non multi-algo Bitcoin members with

`"coin": "ergo",`

**Reply by bdk773** [reply]
> I'd say any of the non multi-algo Bitcoin members with
> 
> `"coin": "ergo",`

Thank you @oliverw again for assistance, I appreciate it.  Excuse my ignorance but could you please elaborate "non multi-algo BTC members" for me?

Lastly for the sake of this thread, would VTC or BTG be a good sample for ERGO?

**Reply by oliverw** [reply]
> > I'd say any of the non multi-algo Bitcoin members with
> > `"coin": "ergo",`
> 
> Thank you @oliverw again for assistance, I appreciate it. Excuse my ignorance but could you please elaborate "non multi-algo BTC members" for me?
> 
> Lastly for the sake of this thread, would VTC or BTG be a good sample for ERGO?

BTC based multi-algo coins are DGB and XVG. VTC should be fine.

**Comment by lurchinms** [comment]
So we are building private stuff now???

**Reply by oliverw** [reply]
Yes.

**Comment by lurchinms** [comment]
That's not Kool......

**Comment by def670** [comment]
this is how i started the docker app in the beginning
sudo docker run -d \    -p 9030:9030 \    -p 127.0.0.1:9053:9053 \    -v /home/def/ergo/.ergo \    -v /home/def/ergo/myergo.conf \    -e MAX_HEAP=3G \    ergoplatform/ergo:v4.0.13 --mainnet -c /home/def/ergo/myergo.conf

thanks for reaching outDon Kingdon

 

    On Wednesday, September 22, 2021, 06:22:03 p.m. EDT, Calvin Tam ***@***.***> wrote:  
 
 


can you share your docker command? it isbl likely you forgot to open the port/misconfig listen ip from miningcore conf

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by calvintam236** [reply]
i don’t think you can use 127.0.0.1:9053. it refers to the 127.0.0.1 inside container. you should have reverse proxy to allow open ports (9030) to pass through and block all open ports in firewall.

**Reply by def670** [reply]
ok i got the node working and its almost done syncing.  miningcore seems to be starting correctly - is there a further frontend that is needed or ?  Im not seeing anything familiar for the generation of a webpage.  Sorry if its a noob question - i am just starting with miningcore - previous experince with open-ethereum-pool 
 
using the full node install - not docker in this build

**Comment by Ravelgijiold** [comment]
Any chance to have an example of ergo.conf file ?

**Reply by szilardkoppel** [reply]
Everything is well documented.

**Reply by def670** [reply]
thanks so much for the totally useful comment - it helped a lot

**Comment by def670** [comment]
ok - im gonna build another vm this weekend and try again.  Ill give that a try - but that was a docker startup line i had found in the forum i believe.  im starting to feel slightly retarded with this node failing to work over and over.  starting to remind me of my first days with open-ethereum-pool

Don Kingdon


    On Thursday, September 23, 2021, 12:10:27 a.m. EDT, Calvin Tam ***@***.***> wrote:  
 
 i don’t think you can use 127.0.0.1:9053. it refers to the 127.0.0.1 inside container. you should have reverse proxy to allow open ports (9030) to pass through and block all open ports in firewall.

**Comment by def670** [comment]
I have tried this:sudo docker run -d \    -p 9030:9030 \    -p 127.0.0.1:9053:9053 \    -v /home/def/ergo/.ergo \    -v /home/def/ergo/myergo.conf \    -e MAX_HEAP=3G \    ergoplatform/ergo:v4.0.13 --mainnet -c /home/def/ergo/myergo.conf
this successfully starts node and api but i dont know how to adjust the config file after setting the APIkey.
with regular node - this fails:
ergo {    directory = ${ergo.directory}"/.ergo"    node {        mining = false    }    wallet.secretStorage.secretDir = ${ergo.directory}"/wallet/keystore"}

this also fails with regular node:
ergo {  node {    mining = true  }}
scorex { restApi {    # Hex-encoded Blake2b256 hash of an API key.     # Should be 64-chars long Base16 string.    # below is the hash of the string 'hello'    # replace with your actual hash     apiKeyHash = "0afa48bb51c2b3c8f9f090922a020a32f244968bda9e7fefeffa68ee3558c584"  }}

**Comment by oliverw** [comment]
Here's an example pool config:

```json
{
  "id": "ergo",
  "enabled": true,
  "coin": "ergo",
  "address": "9foYU8JkoqWBSDA3ba8VHfduPXV2NaVNPPAFkdYoR9t9cPQGMv4",
  "maxActiveJobs": 20,
  "blockRefreshInterval": 1000,
  "jobRebroadcastTimeout": 50,
  "clientConnectionTimeout": 600,
  "rewardRecipients": [
    {
      "address": "9foYU8JkoqWBSDA3ba8VHfduPXV2NaVNPPAFkdYoR9t9cPQGMv4",
      "percentage": 0.5
    }
  ],
  "banning": {
    "enabled": true,
    "time": 180,
    "invalidPercent": 90,
    "checkThreshold": 150
  },
  "ports": {
    "3194": {
      "listenAddress": "0.0.0.0",
      "difficulty": 0.1,
      "name": "GPU Mining",
      "varDiff": {
        "minDiff": 0.01,
        "targetTime": 15,
        "retargetTime": 90,
        "variancePercent": 30,
        "maxDelta": 0.1
      }
    }
  },
  "daemons": [
    {
      "host": "localhost",
      "port": 9052,
      "apiKey": "foobar"
    }
  ],
  "paymentProcessing": {
    "enabled": true,
    "walletPassword": "barfoo",
    "minimumPayment": 0.05,
    "payoutScheme": "PPLNS",
    "payoutSchemeConfig": {
      "factor": 1.0
    }
  }
}

```

**Reply by bdk773** [reply]
@oliverw thank You I appreciate. So it seems that "apikey" and "walletPassword" are the missing pieces or not matching VTC config as it was suggested to use. PS. is the apikey looking for the hash or actual string? I would assume hash since looking at ergo.conf?!

**Reply by oliverw** [reply]
> @oliverw thank You I appreciate. So it seems that "apikey" and "walletPassword" are the missing pieces or not matching VTC config as it was suggested to use. PS. is the apikey looking for the hash or actual string? I would assume hash since looking at ergo.conf?!

Actual unhashed value.

**Reply by bdk773** [reply]
@oliverw one more thing, ergo node default mining port is 9030 where 9053 is to access swagger/api. Your example showing 9052 which is closer to the api backend. Can you confirm which port MC requires (swagger or mining)?

java       875       ergoadmin   13u  IPv6  29094      0t0  TCP *:9030 (LISTEN)
java       875       ergoadmin   30u  IPv6  29101      0t0  TCP *:9053 (LISTEN)

**Reply by bdk773** [reply]
@oliverw may I ask how is the integer in difficulty parameter corresponds to memory size? i.e if you want to allow 4gb and up what would the difficulty look like?!

"difficulty": 0.1,

**Reply by bdk773** [reply]
> 
> 
> @oliverw may I ask how is the integer in difficulty parameter corresponds to memory size? i.e if you want to allow 4gb and up what would the difficulty look like?!
> 
> "difficulty": 0.1,

Anyone knows?
